### DISCUSSION TITLE
Why does pressing p execute paste when filling out snippet?

### DISCUSSION THREAD

**Discussion by pogopaule** [L3MON4D3/LuaSnip#726] [Q&A] [answered]
In the screenshot I just expanded the for-snippet (the cursor is pink)

![image](https://user-images.githubusercontent.com/576949/212768919-810fc3f1-ac27-4cee-a0fb-de4cd8207537.png)

If I press `p` now to enter e.g. the word 'pretty', paste is executed, which I don't want to happen.

If I press any other character I can fill out the blanks as one would expect.

Any idea why this happens? Is this a bug?

**Comment by L3MON4D3** [comment, marked as answer]
Could you check `:map p`? I suspect that some mapping meant for visual only is registered via `vmap` (applies to SELECT and VISUAL) instead of `xmap` (`:h map-table`)

**Comment by pogopaule** [comment]
Oh! Thank you so much for your quick answer. You are right. This is the culprit https://github.com/pogopaule/dotfiles/blob/7f5f7427e9efa637221c7d653593473c9dca229e/nvim/lua/config/keymaps.lua#L40. Changing v to x fixes the issue.

I didn't know the difference between x, v and s...until today

**Reply by L3MON4D3** [reply]
Nice! Yeah, an issue opened here also lead me to discover the distinction, pops up every now and then :D
